pkgs/tools/inputmethods/input-remapper/default.nix

```diff
@@ -0,0 +1,160 @@
+
+    # fix exit code for test script
+    substituteInPlace tests/test.py \
+      --replace 'unittest.TextTestRunner(verbosity=2).run(testsuite)' 'sys.exit(not unittest.TextTestRunner(verbosity=2).run(testsuite).wasSuccessful())'
+
+    # fix test_rename_config
+    # TODO remove after next version
+    substituteInPlace tests/testcases/test_migrations.py \
+      --replace 'os.rmdir(new)' 'import shutil;shutil.rmtree(new)'
+
+    # use build dir not /tmp
+    ${findutils}/bin/find tests -iname '*.py' | while read file; do substituteInPlace "$file" --replace '"/tmp' '"/build/tmp'; done
+  '' + (if xmodmap == null then "" else ''
+    # fix paths for xmodmap
+    substituteInPlace tests/test.py \
+      --replace '"xmodmap"' '"${xmodmap}/bin/xmodmap"'
+    substituteInPlace inputremapper/system_mapping.py \
+      --replace '["xmodmap", "-pke"]' '["${xmodmap}/bin/xmodmap", "-pke"]' \
+      --replace '"Optional `xmodmap` command not found. This is not critical."' '"Optional `xmodmap` command not found. This is not critical.", exc_info=True'
+  '');
+
+  doCheck = true;
+  checkInputs = [
+    procps
+  ] ++ xmodmapList;
+  disabledTests = [ ];
+  pythonImportsCheck = [
+    "evdev"
+    "inputremapper"
+  ];
+
+  # Custom test script, can't use plain pytest / pytestCheckHook
+  # We only run tests which don't need UI or dbus
+  # Skipped
+  # test_daemon (dbus)
+  # test_gui (UI)
+  # test_data (checks for /usr and writable dir, not with nix)
+  # test_injector (not in the sandbox)
+  # test_macros, test_reader, test_keycode_mapper, test_consumer_control (relies on sleeping during tests or checking how long slept for so fails on slow system or high load)
+  installCheckPhase = ''
+    mkdir /build/tmp
+    python tests/test.py \
+      test_config \
+      test_context \
+      test_control \
+      test_dev_utils \
+      test_event_producer \
+      test_groups \
+      test_ipc \
+      test_key \
+      test_logger \
+      test_mapping \
+      test_migrations \
+      test_paths \
+      test_presets \
+      test_user
+  '';
+
+  # Nixpkgs 15.9.4.3. When using wrapGAppsHook with special derivers you can end up with double wrapped binaries.
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    gettext
+    gtk3
+    glib
+    gobject-introspection
+    pygobject3
+  ] ++ xmodmapList;
+
+  propagatedBuildInputs = [
+    setuptools # needs pkg_resources
+    pygobject3
+    evdev
+    pkgconfig
+    pydbus
+    psutil
+  ] ++ xmodmapList;
+
+  postInstall = ''
+    sed -r "s#RUN\+\=\"/bin/input-remapper-control#RUN\+\=\"$out/bin/input-remapper-control#g" -i data/99-input-remapper.rules
+    sed -r "s#ExecStart\=/usr/bin/input-remapper-service#ExecStart\=$out/bin/input-remapper-service#g" -i data/input-remapper.service
+    sed -r "s#WantedBy\=default.target#WantedBy\=graphical.target#g" -i data/input-remapper.service
```

Comment:
bit ambivalent about this one. why not keep `default.target` here and in the module definition?

Reply:
I'll remove this line since it doesn't do anything anyway - WantedBy is ignored in .service files added by packages, only the one in the module does anything.

Leftover from earlier tinkering.